src/read.c

```diff
@@ -2485,10 +2485,7 @@ avifResult avifDecoderReset(avifDecoder * decoder)
                 if (readResult != AVIF_RESULT_OK) {
                     return readResult;
                 }
-                if ((readData.data == NULL) || (readData.size != item->size)) {
```

Comment:
I analyzed the avifDecoderReadItem() code and concluded that this cannot be true if partialByteCount is 0 and avifDecoderReadItem() returns AVIF_RESULT_OK. So we can delete this check.